### PR TITLE
Zoom improvements

### LINKED ISSUE #634: Add keyboard shortcut tooltips to zoom controls
## Problem

It's not obvious you can zoom with Ctrl+Zoom (see #630).

## Proposal

We should make the shortcut more discoverable!

## Prior Art

We show contextual keyboard shortcuts in tooltips throughout elementary OS.

### PR DESCRIPTION
**Pull request by jeremypw**
- Increase possible range of zoom
 - Maintain fine zoom control by holding down `<Alt>` as well as `<Ctrl>`
 - Improve smoothness of scrolling by aggregating smooth scroll events and throttling scroll events
 - Update zoom slider and cursor outside of scroll handler
 - Correct interpolation function
 - Add tooltips to SliderAssembly
 - Use same tooltips in CollectionPage and EditingHostPage, lose custom tooltip property

Fixes #634 

Note that unmodified scrolling also works to zoom when hovering the zoom control (in master) but as this does not work otherwise it has been omitted from the tooltip.

Likewise fine zooming only works when hovering the image in EditingHostPage so has been omitted from the zoom control.

In both places the SliderAssembly is used, the tooltips are now similar to those used in the Code zoom control.  Files uses "zoom" to refer to thumbnail size.

### CODE REVIEW

**Review by cassidyjames — CHANGES_REQUESTED** [review]
One relatively minor thing noted, plus a question: is is possible to use GLib Actions throughout these shortcut tooltips instead of manually specifying the shortcuts in the tooltip copy?

**Comment by jeremypw** [comment]
The existing code relating to actions and accelerators is rather involved :grimacing:  I'll see whether I can introduce a zoom actin

**Comment by jeremypw** [comment]
@cassidyjames  Looks like the existing infrastructure for actions uses deprecated code like Gtk.ActionEntry and Gtk.UIManager.

File: src/Widgets/SliderAssembly.vala [hunk 1]
```diff
@@ -72,14 +68,23 @@ public class SliderAssembly : Gtk.Grid {
 
         add (decrease_box);
 
-        slider = new Gtk.Scale (Gtk.Orientation.HORIZONTAL, null);
+        var primary_tooltip_text = _("Adjust Zoom");
+        var secondary_tooltip_text = _("Control + Scroll, Control + Plus, Control + Minus");
```
[the remark this answers is not in the document]
Comment (jeremypw) [reply src/Widgets/SliderAssembly.vala]: Done

**Comment by jeremypw** [comment]
I think any refactoring to use actions would be too extensive for this PR.  Maybe the whole action infrastructure can be reviewed to remove deprecated code in another PR.

**Review by cassidyjames — APPROVED** [review]

**Pull request merged**